combat-trainer.lic

```diff
@@ -1287,6 +1358,30 @@ class SpellProcess
     end
   end
 
+  def check_trader_magic(game_state)
+    return unless DRStats.trader?
+    if Flags['ct-starlight-depleted'] then
+      echo "----OUT OF STARLIGHT - DELETING STARLIGHT MAGIC----"
+      @buff_spells.delete_if { |spell| @buff_spells[spell]['starlight'] }
+      @offensive_spells.delete_if do |spell|
+        if spell['starlight'] then true end
```

Comment:
💡 Could this be `@offensive_spells.delete_if { |spell| spell['starlight'] }`?

Reply:
Sort of.  It's surprisingly different from buff_spells for whatever reason.  Switched it over to reject but kept the format it needed to do its thing.